<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditTableDialog</class>
 <widget class="QDialog" name="EditTableDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>414</width>
    <height>405</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit table definition</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons/icons.qrc">
    <normaloff>:/icons/table</normaloff>:/icons/table</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="groupTable">
     <property name="title">
      <string>Table</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLineEdit" name="editTableName">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupFields">
     <property name="title">
      <string>Fields</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QToolButton" name="addFieldButton">
          <property name="text">
           <string>Add field</string>
          </property>
          <property name="icon">
           <iconset resource="icons/icons.qrc">
            <normaloff>:/icons/field_add</normaloff>:/icons/field_add</iconset>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="renameFieldButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Edit field</string>
          </property>
          <property name="icon">
           <iconset resource="icons/icons.qrc">
            <normaloff>:/icons/field_edit</normaloff>:/icons/field_edit</iconset>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="removeFieldButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Remove field</string>
          </property>
          <property name="icon">
           <iconset resource="icons/icons.qrc">
            <normaloff>:/icons/field_delete</normaloff>:/icons/field_delete</iconset>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTreeWidget" name="treeWidget">
        <column>
         <property name="text">
          <string>Field name</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Field type</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>editTableName</tabstop>
  <tabstop>addFieldButton</tabstop>
  <tabstop>renameFieldButton</tabstop>
  <tabstop>removeFieldButton</tabstop>
  <tabstop>treeWidget</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources>
  <include location="icons/icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EditTableDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>252</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EditTableDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidget</sender>
   <signal>itemSelectionChanged()</signal>
   <receiver>EditTableDialog</receiver>
   <slot>fieldSelectionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>116</x>
     <y>193</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addFieldButton</sender>
   <signal>clicked()</signal>
   <receiver>EditTableDialog</receiver>
   <slot>addField()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>renameFieldButton</sender>
   <signal>clicked()</signal>
   <receiver>EditTableDialog</receiver>
   <slot>editField()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>161</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeFieldButton</sender>
   <signal>clicked()</signal>
   <receiver>EditTableDialog</receiver>
   <slot>removeField()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>276</x>
     <y>111</y>
    </hint>
    <hint type="destinationlabel">
     <x>249</x>
     <y>63</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editTableName</sender>
   <signal>textChanged(QString)</signal>
   <receiver>EditTableDialog</receiver>
   <slot>checkInput()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>62</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>115</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidget</sender>
   <signal>itemChanged(QTreeWidgetItem*,int)</signal>
   <receiver>EditTableDialog</receiver>
   <slot>checkInput()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>fieldSelectionChanged()</slot>
  <slot>addField()</slot>
  <slot>editField()</slot>
  <slot>removeField()</slot>
  <slot>checkInput()</slot>
 </slots>
</ui>
